src/lib.rs

```diff
@@ -684,6 +694,50 @@ pub trait Context {
     unsafe fn stencil_op(&self, stencil_fail: u32, depth_fail: u32, pass: u32);
 
     unsafe fn stencil_op_separate(&self, face: u32, stencil_fail: u32, depth_fail: u32, pass: u32);
+
+    unsafe fn debug_message_control(
+        &self,
+        source: u32,
+        msg_type: u32,
+        severity: u32,
+        ids: Option<&[u32]>,
+        enabled: bool,
+    );
+
+    unsafe fn debug_message_insert<S>(
+        &self,
+        source: u32,
+        msg_type: u32,
+        id: u32,
+        severity: u32,
+        msg: S,
+    ) where
+        S: AsRef<str>;
+
+    unsafe fn debug_message_callback(
+        &self,
+        callback: Self::DebugProc,
+        user_data: *mut std::ffi::c_void,
+    );
+
+    unsafe fn get_debug_message_log(
+        &self,
+        count: u32,
+    ) -> Vec<DebugMessageLogEntry>;
+
+    unsafe fn get_pointerv(&self, pname: u32, params: *const *mut std::ffi::c_void);
```

Comment:
Could we avoid the use of raw pointers somehow? i.e. providing callbacks with `impl Fn` or similar

Reply:
Can you clarify how this would work? I'm not sure how much more type safety we could give this as `glGetPointerv` can return a ton of different things

Reply:
I was thinking we could avoid exposing`glGetPointerv` and raw pointers for user data/callbacks and have our own typed functions/structs instead.

Similar to the deprecated WebGL proposal, we would have `debug_object_label` and `get_debug_object_label` accepting our own struct (the struct can still carry a pointer internally if necessary).

Callbacks can accept typed functions instead of pointers, and the typed functions can call the raw callback internally.